primitives/src/migrate.rs

```diff
@@ -87,6 +87,37 @@ pub fn migrate_map_rename<T: Migrate, C: FnMut(&[u8]) -> T::Context>(
         .for_each(|(key, new)| put_storage_value(module, new_item, &key, new));
 }
 
+/// Migrate the key & value of a map `KO, VO` to key & value of type `KN, VN` via `map`.
+/// The map is located in `module::item` and the new map will migrate to `module::new_item`.
+/// This assumes that the hashers are all the same, which is the common case.
+pub fn migrate_map_keys_and_value<VO, VN, H, KO, KN, F>(
+    module: &[u8],
+    item: &[u8],
+    new_item: &[u8],
+    mut map: F,
+) where
+    F: FnMut(KO, VO) -> (KN, VN),
+    VO: Decode + Encode,
+    VN: Encode + Decode,
+    H: ReversibleStorageHasher,
+    KO: Decode,
+    KN: Decode + Encode,
+{
+    let old_map = StorageIterator::<VO>::new(module, item)
```

Comment:
`StorageKeyIterator` would be better in that it does reversing and decoding internally.